frontend/src/store/modules/geo_location/actions.js

```diff
@@ -1,11 +1,35 @@
-import {CHANGE_SELECTED_PERIOD, CHANGE_SELECTED_PARAMETER} from "./types/actions";
-import {SET_SELECTED_PERIOD, SET_SELECTED_PARAMETER} from "./types/mutations";
+import {
+    CHANGE_SELECTED_PERIOD,
+    CHANGE_SELECTED_PARAMETER,
+    FETCH_GEO_LOCATION_ITEMS
+} from "./types/actions";
+import {
+    SET_SELECTED_PERIOD,
+    SET_SELECTED_PARAMETER,
+    SET_GEO_LOCATION_ITEMS,
+    RESET_GEO_LOCATION_ITEMS
+} from "./types/mutations";
+import {getGeoLocationItems} from '@/api/geo_location/geoLocationItemsService';
+import {getTimeByPeriod} from '@/services/periodService';
 
 export default {
     [CHANGE_SELECTED_PERIOD]: (context, payload) => {
         context.commit(SET_SELECTED_PERIOD, payload.value);
+        context.dispatch(FETCH_GEO_LOCATION_ITEMS);
     },
     [CHANGE_SELECTED_PARAMETER]: (context, parameter) => {
         context.commit(SET_SELECTED_PARAMETER, parameter);
+    },
+    [FETCH_GEO_LOCATION_ITEMS]: context => {
```

Comment:
Don't forget to toggle spinner on and off during this action, since you need to show user that something is happening